test/kernel_apps.js

```diff
@@ -46,16 +52,22 @@ contract('Kernel apps', accounts => {
     it('fails if initializing on constructor before setting app code', async () => {
         const initializationPayload = code1.contract.initialize.getData()
 
-        return assertRevert(async () => {
-            await AppProxyUpgradeable.new(kernel.address, appId, initializationPayload)
-        })
+        try {
+            let result = await AppProxyUpgradeable.new(kernel.address, appId, initializationPayload, { gas: 5e6 })
+        } catch (e) {
+            if (codeDeploymentError(e)) { return true }
+            assert.isAbove(e.message.search('revert'), -1, 'should have failed with revert')
+        }
```

Comment:
It'd be nice to hoist this up as a `assertCodeDeploymentErrorOrRevert`:

```js
const assertCodeDeploymentErrorOrRevert = code => {
  return assertRevert(async () => {
    try {
      await code()
    } catch (e) {
      if (codeDeploymentError(e)) { throw new Error('mock revert') }
      else { throw e }
    }
  })
}
```

Reply:
By "hoist" do you mean add this as function in `helpers/assertThrow.js`?